protocol/messenger_mailserver.go

```diff
@@ -288,7 +288,7 @@ func (m *Messenger) syncFilters(filters []*transport.Filter) (*MessengerResponse
 	prioritizedBatches := getPrioritizedBatches()
 	currentBatch := 0
 
-	if filters[0].Priority == 0 {
+	if len(filters) == 0 || filters[0].Priority == 0 {
```

Comment:
I would need to read some golang specific docs to be 100% sure but usually the order of evaluation of `or` is so that the second expression is skipped when the first one is equal to `true`. And I suppose that should work in our case. 
So if  `len(filters) == 0` (which means that `filters` is empty), then the second shouldn't be executed. Otherwise we are safe to use `filters[0].Priority == 0`.

Reply:
aaah okay. I am new to golang, I think in C++ it would fail. approving it :)

Reply:
@Khushboo-dev-cpp quick googling showed that it works exactly the same way in c++ (I mean skipping of the rest of `or` when the first expression is `true`), I bet it is going to be a challenge to find some more or less useful language which has a different evaluation order for this case. Well, maybe I'm wrong... :)